### PR TITLE
[REG-1279 v2] Add new in-depth getting started tutorial

### PR DESCRIPTION
**Pull request by vontell**
This PR has been a long time in the making 😬 it updates the getting started doc to finally work with our most recent changes, such as the new state changes, new SDK process, and new agent builder pieces. 

I'll admit, I did take a few shortcuts here. For instance, I removed a section I wanted to add for editing this in agent builder,  and they may find that editing this with agent builder would break an import. I wanted to at least get this so we have something that is far better than we had before in the docs (the current getting started guide doesn't work at all), but there may be follow-ups depending on priorities.

This also relies on a game sample and marketplace bot, found here:
https://github.com/Regression-Games-Marketplace/SampleDemoScene
https://github.com/Regression-Games-Marketplace/Simple-Kiting-Bot

The only remaining piece here is that when we deploy, we need to upload and publish this bot. That will be in a follow up PR once we push these to prod.
Screenshot 2024-02-07 at 4 27 41 PM
Screenshot 2024-02-07 at 4 01 49 PM

### CODE REVIEW

File: docs/tutorials/building-your-first-bot.mdx [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+    }
+
+}
+```
+
+Once this component is implemented, create an empty object within the scene, rename it to `RGBotSpawner` (just so we can find it later),
+and attach this `CharacterBotSpawnManager` component to that object. Position it within the scene wherever you want to spawn your bot 
+(**make sure to set the Y coordinate to 0 so your bot does not float!**), and drag `RGBotSpawner`'s own Transform into the 
+**Bot Spawn Point** entry of the RGBotSpawner's `CharacterBotSpawnManager` component. Finally, find the 
+**Player** prefab provided by the sample project (located within `Assets/FirstBotDemo/Demo/Prefabs/Player.prefab`), and drag
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:163]: Let's avoid parenthesis for important instructions like this.

```suggestion
**Player** prefab under `Assets > FirstBotDemo > Demo > Prefabs` and drag 
```

File: docs/tutorials/building-your-first-bot.mdx [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+
+}
+```
+
+Once this component is implemented, create an empty object within the scene, rename it to `RGBotSpawner` (just so we can find it later),
+and attach this `CharacterBotSpawnManager` component to that object. Position it within the scene wherever you want to spawn your bot 
+(**make sure to set the Y coordinate to 0 so your bot does not float!**), and drag `RGBotSpawner`'s own Transform into the 
+**Bot Spawn Point** entry of the RGBotSpawner's `CharacterBotSpawnManager` component. Finally, find the 
+**Player** prefab provided by the sample project (located within `Assets/FirstBotDemo/Demo/Prefabs/Player.prefab`), and drag 
+this into the **Rg Bot Prefab** entry. Regression Games now knows how to load and unload bots from your scene! Make sure to save the
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:164]:
```suggestion
it into the **Rg Bot Prefab** entry. Regression Games now knows how to load and unload bots from your scene! Make sure to save the 
```

File: docs/tutorials/building-your-first-bot.mdx [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+this into the **Rg Bot Prefab** entry. Regression Games now knows how to load and unload bots from your scene! Make sure to save the 
+scene.
+
+![Screenshot of the RGOverlayCanvas.](./img/building-your-first-bot/spawner_6.png)
+
+## Add state information to relevant GameObjects
+
+We are onto the final parts of the integration! We need to indicate the following within our game:
+
+* What game state our bots can see
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:173]: Thinking that "game state" might not make sense to someone unfamiliar with our concepts. Also I don't think we need bullets if there's only two of them. Suggestion:

We are onto the final parts of the integration! We now need to indicate what information the bot has access to in the game (referred to here as the game's "state"), and which actions the bot can perform.

File: docs/tutorials/building-your-first-bot.mdx [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+* What actions our bots can perform
+
+First, let's add the state information. Regression Games marks states for bot consumption using the
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:176]:
```suggestion
First, let's define some state information. Regression Games marks states for bot consumption using the 
```

Comment (abeizer) [reply docs/tutorials/building-your-first-bot.mdx:176]: And for following sentence, suggestion:

"Regression Games uses the `[RGState]` attribute to designate which state information bots can consume"

File: docs/tutorials/building-your-first-bot.mdx [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+
+We are onto the final parts of the integration! We need to indicate the following within our game:
+
+* What game state our bots can see
+* What actions our bots can perform
+
+First, let's add the state information. Regression Games marks states for bot consumption using the 
+[`[RGState]` attribute](/integrating-with-unity/defining-states#3---rgstate-fieldpropertymethod-attribute).
+This attribute can be applied to any public method that returns a value or any variable within a `MonoBehaviour`, and
+it will automatically be sent to the bot.
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:179]: Maybe instead, (new sentence) "Any values designated this way are automatically passed to your bots."

File: docs/tutorials/building-your-first-bot.mdx [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+
+First, open the `Assets/FirstBotDemo/Demo/Scripts/Player/PlayerAttack.cs` file, and add the following method to the class, which
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:181]: You already use "First" above. I think you can just combine this paragraph with the previous one

Comment (abeizer) [reply docs/tutorials/building-your-first-bot.mdx:181]: I also see that you're mixing how you define paths - sometimes you use `>` and sometimes `/`. Let's pick one or the other (I personally prefer the `>` because I think it looks nice)

Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:181]:
```suggestion
First, open the `Assets/FirstBotDemo/Demo/Scripts/Player/PlayerAttack.cs` file, and add the following method to the class. This
```

File: docs/tutorials/building-your-first-bot.mdx [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+```
+
+Then, within the `Assets/FirstBotDemo/Demo/Scripts/EnemyController.cs` file, add the following attributes to the existing `GetCurrentHp()`
+and `GetTotalHp()` methods:
+
+```csharp
+/*
+* Gets the enemies current hp
+*/
+[RGState("CurrentHealth")]
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:200]: I see the value in showing users that they can override a state name, but that isn't explained here so this looks confusing. Why not just call the methods `CurrentHealth()` and `MaxHealth()` for the purpose of the tutorial?

File: docs/tutorials/building-your-first-bot.mdx [hunk 8]
```diff
@@ -0,0 +1,342 @@
...
+    {
+        Debug.LogWarning($"{gameObject.name} is missing entity info. Please check the inspector and assign missing fields");
+        return default;
+    }
+    return enemyInfo.hp;
+}
+```
+
+That's it! The bot will now be able to see the current and max
+health of the enemy, and whether or not the player is attacking. See the full [state documentation](/integrating-with-unity/defining-states)
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:222]: I think reverse these

"The bot will now be able to tell when it is attacking the enemy, and what the enemy's current and max health values are."

File: docs/tutorials/building-your-first-bot.mdx [hunk 9]
```diff
@@ -0,0 +1,342 @@
...
+to see all the ways you can define state within your game, including grabbing all of the state from a MonoBehaviour.
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:223]:
```suggestion
for more information on defining states within your game.
```

Concerned that the "including grabbing all of the state from a MonoBehaviour" won't make sense to someone at this point in the tutorial since they haven't gotten to the point of actually _using_ states for anything.

File: docs/tutorials/building-your-first-bot.mdx [hunk 10]
```diff
@@ -0,0 +1,342 @@
...
+
+## Implement actions your bot can take
+
+Similar to states, we will now tag the actions that a bot can take, using [`[RGAction]`](/integrating-with-unity/defining-actions). In our case, a bot can either move or attack.
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:227]:
```suggestion
Similarly to states, we will now define actions that the bot can take using the [`[RGAction]`](/integrating-with-unity/defining-actions) attribute. In our case, a bot can either move or attack.
```

File: docs/tutorials/building-your-first-bot.mdx [hunk 11]
```diff
@@ -0,0 +1,342 @@
...
+
+Open the `Assets/FirstBotDemo/Demo/Scripts/Player/PlayerInputControl.cs` file, and replace the contents with the following.
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:229]:
```suggestion
Open the `Assets/FirstBotDemo/Demo/Scripts/Player/PlayerInputControl.cs` file, and replace its contents with the following.
```

File: docs/tutorials/building-your-first-bot.mdx [hunk 12]
```diff
@@ -0,0 +1,342 @@
...
+This change does a few things:
+
+1. Adds an editor field that allows us to mark a specific prefab instance as being a bot (vs a character controlled by a real player).
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:232]:
```suggestion
1. Adds an editor field that allows us to mark a specific prefab instance as being controlled by a bot (versus a real player).
```

File: docs/tutorials/building-your-first-bot.mdx [hunk 13]
```diff
@@ -0,0 +1,342 @@
...
+2. Uses that field to ensure that keyboard/input handling only applies to non-bot game objects.
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:233]:
```suggestion
2. Ensures that physical inputs from mouse and keyboard are ignored by bots (so that you can play alongside your bot without interfering with its movements)
```

Do we need to describe this at all though? This tutorial doesn't allow a bot + player combo so it's not relevant to the sample

File: docs/tutorials/building-your-first-bot.mdx [hunk 14]
```diff
@@ -0,0 +1,342 @@
...
+3. Marks the movement method as being an action that the bot can perform.
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:234]: Designates? "Marks" doesn't sound right to me, but I'm not sure what other word I'd use

File: docs/tutorials/building-your-first-bot.mdx [hunk 15]
```diff
@@ -0,0 +1,342 @@
...
+        
+        public void JumpInput(bool newJumpState)
+        {
+            jump = newJumpState;
+        }
+    }
+}
+```
+
+Then, open the `Assets/FirstBotDemo/Demo/Scripts/Player/PlayerAttack.cs` file again, and add the following method to the class,
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:277]:
```suggestion
Then, open the `Assets/FirstBotDemo/Demo/Scripts/Player/PlayerAttack.cs` file again, and add the following method to the class.
```

File: docs/tutorials/building-your-first-bot.mdx [hunk 16]
```diff
@@ -0,0 +1,342 @@
...
+which defines an action that allows a bot to target an enemy and use an ability:
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:278]:
```suggestion
This defines an action that allows the bot to target an enemy and use an ability on that enemy:
```

File: docs/tutorials/building-your-first-bot.mdx [hunk 17]
```diff
@@ -0,0 +1,342 @@
...
+    SelectEnemy(enemy);
+    Attack(abilities[ability]);
+}
+```
+
+
+As a last step here, in the player input code, you will see that we introduced an `isBot` field to the `PlayerInputControl` component.
+This will allow our player to be controlled by the bot logic when it is set to true. However, there is an existing `Player`
+GameObject in the scene, which we want to control using our manual input controls. Click this `Player` in the object hierarchy,
+find the `Player Input Control` component, set `Is Bot` to unchecked, and then save (you may find that it is already unchecked).
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:294]: Since we're doing all of this in the context of bots rather than human players, I think it makes more sense to default `isBot` to `true` and have the user check the box on their bot prefab. Once you started talking about the `Player` object, I was confused about how the Player is used in this tutorial, but it looks like they aren't - this step is just to make sure the Player doesn't "break" from these changes

File: docs/tutorials/building-your-first-bot.mdx [hunk 18]
```diff
@@ -0,0 +1,342 @@
...
+
+That's it for actions! A bot may now call into two actions (moving and attacking), which we will demonstrate in the section
+when we build our bot.
+
+## Using a ready-to-go bot from the Bot Marketplace
+
+Now that our integration is complete, our final step is to use an implementation of a bot. We will start with a template
+found in our [Bot Marketplace](https://play.regression.gg/marketplace), and then will show you where to edit its code.
+
+Go to [this page](https://play.regression.gg/marketplace/1) to see the existing bot, and click "Create Bot" to create a copy
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:306]: Give the link "meaning" -> 
Find the [Simple Kiting Bot in the Bot Marketplace](https://play.regression.gg/marketplace/1) and click "Create Bot" to copy it to your Regression Games account.

File: docs/tutorials/building-your-first-bot.mdx [hunk 19]
```diff
@@ -0,0 +1,342 @@
...
+
+![Screenshot of the agent builder configuration for this bot.](./img/building-your-first-bot/agent_builder.png)
+
+Go back to Unity, and within the **Regression Games** menu in the menu bar, click **Synchronize Bots with RG**. This will pull your new
+bot into your Unity project. 
+
+![Screenshot of the agent builder configuration for this bot.](./img/building-your-first-bot/synchronize_bots.png)
+
+At this point, you can click play, and then select the Regression Games overlay button in the bottom
+right of your game screen. Use the dropdown to select the bot you just created, and then click "Start Bot". The bot will begin to
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:322]:
```suggestion
right corner of your Scene. Use the dropdown to select the bot you just created, and then click "Start Bot". The bot will begin to
```

File: docs/tutorials/building-your-first-bot.mdx [hunk 20]
```diff
@@ -0,0 +1,342 @@
...
+run around and shoot the enemy!
+
+![Screenshot of the bot creation page.](./img/building-your-first-bot/start_bot_10.png)
+
+You now have a working bot, fully integrated into your game! If you want to modify the behavior of the bot, you can make edits
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:327]:
```suggestion
You now have a working bot, fully integrated into your game! If you want to modify the bot's behavior, you can make edits
```

File: docs/tutorials/building-your-first-bot.mdx [hunk 21]
```diff
@@ -0,0 +1,342 @@
...
+within the Agent Builder, or edit the code in your Unity project at `Assets/RegressionGames/Runtime/Bots/BOTNAME/Nodes`
+
+## Next Steps
+
+Congratulations, you've now experienced the integration process for bots! Using this guide and our reference material,
+you can now integrate bots into your own game. If you have any questions, please reach out to us on our 
+[Discord](https://discord.com/invite/925SYVse2H). Please note that we offer **dedicated support** for your integration,
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:334]:
```suggestion
[Discord](https://discord.com/invite/925SYVse2H). Please note that we offer **dedicated support** for your integration. 
```

File: docs/tutorials/building-your-first-bot.mdx [hunk 22]
```diff
@@ -0,0 +1,342 @@
...
+
+You now have a working bot, fully integrated into your game! If you want to modify the behavior of the bot, you can make edits
+within the Agent Builder, or edit the code in your Unity project at `Assets/RegressionGames/Runtime/Bots/BOTNAME/Nodes`
+
+## Next Steps
+
+Congratulations, you've now experienced the integration process for bots! Using this guide and our reference material,
+you can now integrate bots into your own game. If you have any questions, please reach out to us on our 
+[Discord](https://discord.com/invite/925SYVse2H). Please note that we offer **dedicated support** for your integration, 
+we are happy to do it for you or walk you through it on a call step by step.
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:335]:
```suggestion
We are happy to work directly with your project for you, or walk you step-by-step through the integration process over a call.
```

**Review by abeizer — CHANGES_REQUESTED** [review]
Your content is solid, I don't think I'd add or remove much. My comments are mostly rewords, making things flow better, and attempting to clarify instructions for new users. I really recommend taking some time to read through your guide aloud to yourself - this is what I do to identify problematic sentences.

**Comment by RG-nAmKcAz** [comment]
(non-blocking)
Apparently we have a deficiency where we aren't filling in the `botCreationTool` for AgentBuilder bots.
https://github.com/Regression-Games-Marketplace/Simple-Kiting-Bot/blob/main/BotRecord.asset

**Review by RG-nAmKcAz — CHANGES_REQUESTED** [review]
I still need to review the content details, but I have a concern with the fact that we have duplicated pieces/parts of the Sample from the SDK into another GitHub repo again. 

I feel like we need to better align this guide to use the sample, or include this 'tutorial' as an SDK sample itself.

I mention this because I started my review by loading up the doc site on this branch.   

Our quick start page talks about creating a new project, getting the SDK, and loading the sample.  Great, easy, simple and I'm ready to go.  

Then you click 'Next' to go into the Building Your First Bot page (this PR) and I expected it to pickup from where the quickstart left off and give me more detailed information about the ingeration.

Instead the Building Your First Bot page (this PR) has no context / recognition that I just came from the quick start page and tells me to go install some other new project and follow a different setup pattern/sequence-of-steps compared to the quick start guide on the previous page.

The experience felt disconnected and was quite jarring.

File: docs/tutorials/building-your-first-bot.mdx [hunk 23]
```diff
@@ -0,0 +1,342 @@
...
+    {
+        [Header("Character Input Values")]
+        public Vector2 move;
+        public Vector2 look;
+        public bool jump;
+
+        [Header("Movement Settings")]
+        public bool analogMovement;
+
+        [Header("BotSettings")] public bool isBot = true;
```
Comment (RG-nAmKcAz) [inline docs/tutorials/building-your-first-bot.mdx:252]: We should add line comments to this file that correlate the #1,2,3 bullets above to which lines in the code achieve each bullet.

File: docs/tutorials/building-your-first-bot.mdx (hunk 17 above)
Comment (RG-nAmKcAz) [inline docs/tutorials/building-your-first-bot.mdx:294]: set `Is Bot` to `false` (unchecked).  These are programmers/developers, they would think of this field as the boolean it is.

File: docs/tutorials/building-your-first-bot.mdx [hunk 24]
```diff
@@ -0,0 +1,342 @@
...
+
+
+As a last step here, in the player input code, you will see that we introduced an `isBot` field to the `PlayerInputControl` component.
+This will allow our player to be controlled by the bot logic when it is set to true. However, there is an existing `Player`
+GameObject in the scene, which we want to control using our manual input controls. Click this `Player` in the object hierarchy,
+find the `Player Input Control` component, set `Is Bot` to unchecked, and then save (you may find that it is already unchecked).
+
+![Screenshot of the player prefab with the components added.](./img/building-your-first-bot/is_bot.png)
+
+That's it for actions! A bot may now call into two actions (moving and attacking), which we will demonstrate in the section
```
Comment (RG-nAmKcAz) [inline docs/tutorials/building-your-first-bot.mdx:298]: `call into` sounds weird.

Maybe reword this to convey that the bot now has 2 available actions it can invoke (moving or attacking).

File: docs/tutorials/building-your-first-bot.mdx [hunk 25]
```diff
@@ -0,0 +1,342 @@
...
+when we build our bot.
```
Comment (RG-nAmKcAz) [inline docs/tutorials/building-your-first-bot.mdx:299]: `which we will demonstrate in the section when we build our bot`... Where is this section ? 

The next section talks about using a ready to go bot, but does not communicate or reference back to anything about the states/actions we just spent time creating.

**Review by RG-nAmKcAz — CHANGES_REQUESTED** [review]
Core flow of the tutorial is good.

I have some re-wording suggestions in addition to @abeizer 's, as well as some content suggestions.

My biggest concern though is still what I mentioned above in the PR that this guide doesn't fit well with our existing quick start.  I feel like the process we have in the quick start is more 'normal' where you are adding our SDK/sample into your project and working from there.  This tutorial where you start from our project, but then have to still add our SDK into it doesn't feel like the right Unity flow to me.

File: docs/tutorials/building-your-first-bot.mdx [hunk 26]
```diff
@@ -0,0 +1,342 @@
...
+This tutorial assumes a basic understanding of Unity concepts. If you get stuck, join our [Discord](https://discord.com/invite/925SYVse2H) and we can help you out!
+
+:::
+
+In this tutorial, we will get a simple bot integrated and running in a game. The bot will run to a random position and then
+shoot the enemy in our playground sample. This tutorial will cover:
+
+- How to integrate the RG Unity SDK into your game
+- How to add various state and actions attributes so your bot can understand the game
+- How to use a ready-to-go bot from the Bot Marketplace, and how to modify its code
```
[the remark this answers is not in the document]
Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx:18]: Not really official or not here - this was just more of an explanation vs trying to name it something specific. Happy to reword if you have an idea!

File: docs/tutorials/building-your-first-bot.mdx [hunk 27]
```diff
@@ -0,0 +1,342 @@
...
+:::
+
+You can find this prefab by searching in the project file window for `RGOverlayCanvas` (make sure the "In Packages" filter is selected),
+or by navigating to the **Packages** > **Regression Games Unity Bots** > **Runtime** > **Prefabs** folder. Drag this overlay onto your 
+scene in the object hierarchy.
+
+![Screenshot of the search pane for the RGOverlayCanvas.](./img/building-your-first-bot/overlay_4.png)
+
+Once you have this overlay available, you can click play to see this in action! A
+Regression Games logo will appear at the bottom of your screen. Click this button to see a dropdown for your bots. You
```
[the remark this answers is not in the document]
Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx:83]: I think screen is correct here? The scene is the actual game (i.e. bottom of the scene would be the ground?)

File: docs/tutorials/building-your-first-bot.mdx (hunk 10 above)
Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx:227]: From my research:

"Similar To” is the correct expression. “Similarly To” is grammatically incorrect and should be avoided. You may use “Similarly” by itself, with no preposition to accompany it, if you’d like. But it’d be either “Similar To” or just “Similarly”, never “Similarly To”.

File: docs/tutorials/building-your-first-bot.mdx (hunk 19 above)
Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx:322]: Same comment as above regarding scene vs screen - I do mean the screen here. Unless viewport is preferred? I suppose according to unity, "Scene View" is the most correct term, so I can change it to that.

File: docs/tutorials/building-your-first-bot.mdx (hunk 22 above)
Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx:335]: Maybe "work directly on your project for you?" I was thinking with your project sounds weird since we aren't working with the project, we are working on it. (as I type this, grammarly is trying to correct "with" to "on" as well)

File: docs/tutorials/building-your-first-bot.mdx (hunk 18 above)
Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx]: Fixed

File: docs/tutorials/building-your-first-bot.mdx (hunk 19 above)
Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx]: Fixed

File: docs/tutorials/building-your-first-bot.mdx [hunk 28]
```diff
@@ -0,0 +1,342 @@
...
+        // Stops all bots and cleans up the RG SDK
+        RGBotServerListener.GetInstance()?.StopGame();
+    }
+
+}
+```
+
+Once this component is implemented, create an empty object within the scene, rename it to `RGBotSpawner` (just so we can find it later),
+and attach this `CharacterBotSpawnManager` component to that object. Position it within the scene wherever you want to spawn your bot 
+(**make sure to set the Y coordinate to 0 so your bot does not float!**), and drag `RGBotSpawner`'s own Transform into the
```
[the remark this answers is not in the document]
Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx]: I think dragging and dropping either is fine! I would need to try it out... in the meantime, fixed the long sentence issue. We could probably do it the code-only way by getting the game object, but this is written this way because it lends easier to existing games where the spawn point is already provided, and they may just be following along with their own game code.

File: docs/tutorials/building-your-first-bot.mdx [hunk 29]
```diff
@@ -0,0 +1,342 @@
...
+public int GetCurrentHp()
+{
+    return _currentHp;
+}
+
+/*
+* Gets the enemy's total HP from its assigned info
+*/
+[RGState("MaxHealth")]
+public int GetTotalHp()
```
[the remark this answers is not in the document]
Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx:217]: I was hoping to avoid changing the original code, but maybe I can... I was about to respond to this saying we have examples in our RGState code, but I realize we don't seem to have examples in the reference for these difference options. We should likely add that at some point as well...

File: docs/tutorials/building-your-first-bot.mdx (hunk 23 above)
Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx:259]: Added!

File: docs/tutorials/building-your-first-bot.mdx (hunk 17 above)
Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx]: Maybe I can remove all of this because from all of my tests so far, no change is actually needed. It seems like the defaults work as expected.

Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx]: Fixed!

**Review by vontell — COMMENTED** [review]
Adding comments for myself on what I need to do next.

- [ ] Edit the original code to support a property/field RGState for CurrentHealth

File: docs/tutorials/building-your-first-bot.mdx (hunk 29 above)
[the remark this answers is not in the document]
Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx:213]: @RG-nAmKcAz I couldn't find this in the docs, but does the field need to be public to be accessible with the new state approach? If you see my new code above on line 203, I added the attribute to a private field - I still need to test but figured I'd ask here.

File: docs/tutorials/building-your-first-bot.mdx [hunk 30]
```diff
@@ -0,0 +1,346 @@
...
+}
+```
+
+Then, within the `Assets/FirstBotDemo/Demo/Scripts/EnemyController.cs` file, add the following attributes to the existing `GetCurrentHp()`
+and `GetTotalHp()` methods:
+
+```csharp
+// Near line 18
+[RGState("CurrentHealth")]
+private int _currentHp;
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:204]: Sounds like this needs to be public now

File: docs/tutorials/building-your-first-bot.mdx [hunk 31]
```diff
@@ -0,0 +1,346 @@
...
+
+## Add the RGOverlayCanvas
+
+The RGOverlayCanvas prefab provides a drag and drop overlay that lets you easily start and stop bots running
+in your scene. This is extremely useful when initially creating and debugging your bots.
+
+:::caution
+
+Note that this overlay **must be present in your scene** for Regression Games to function. The visibility
+of the overlay can be hidden through the settings. If you have multiple scenes where a bot will be used,
```
Comment (abeizer) [inline docs/tutorials/building-your-first-bot.mdx:75]:
```suggestion
of the overlay can be hidden through its settings. If you have multiple scenes where a bot will be used,
```

**Review by abeizer — CHANGES_REQUESTED** [review]

File: docs/tutorials/building-your-first-bot.mdx (hunk 26 above)
[the remark this answers is not in the document]
Comment (abeizer) [reply docs/tutorials/building-your-first-bot.mdx:18]: This is fine for the doc as-is, I was thinking about what we might want users to know this concept as in the future. Could be generic with "Premade Bot", "Community Bot", "Quickstart Bot", etc. Or we could be one of those companies who uses fun terminology to describe pieces of their product - like anything that can be used interchangeably with our different tools/SDKs could be a "Neuron" or something along those lines... I guess making Marketplace Bots "Pre-Trained Neurons"? Mind wandering now

**Review by RG-nAmKcAz — APPROVED** [review]
I'm going to mark this approved pending Abby's remaining feedback being addressed.  The page definitely flows better from the quickstart before it now and the overall flow / consistency updates have been made.

File: docs/tutorials/building-your-first-bot.mdx (hunk 30 above)
Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx:204]: @RG-nAmKcAz not sure if you saw my comment about this in the PR - would this work? Or does this need to be public? I wasn't sure with the new changes.

File: docs/tutorials/building-your-first-bot.mdx (hunk 31 above)
Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx:75]: I think it should be `the`? It's not the overlay's settings - it's the settings of the RG SDK.

File: docs/tutorials/building-your-first-bot.mdx (hunk 30 above)
Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx:204]: I see now, I missed the public part in the docs.

File: docs/tutorials/building-your-first-bot.mdx (hunk 31 above)
Comment (abeizer) [reply docs/tutorials/building-your-first-bot.mdx:75]: I'd make that explicit then - I thought this was talking about hiding the component from its GameObject. If there's another setting then we should mention where that is

File: docs/tutorials/building-your-first-bot.mdx (hunk 30 above)
Comment (vontell) [reply docs/tutorials/building-your-first-bot.mdx]: Let me know what you think of this... I think having them change it to public is good so they learn that it has to be, but also now the naming scheme is inconsistent with the `_`. Maybe that's ideal/best practice? i.e. maybe that means "this is public but only because RG wants to see it - treat it as private!" Not sure... looking for opinions.

**Comment by vontell** [comment]
@abeizer back to you!

File: docs/tutorials/building-your-first-bot.mdx (hunk 30 above)
Comment (abeizer) [reply docs/tutorials/building-your-first-bot.mdx]: I don't like the idea of asking people to either violate naming conventions or rename the variable and update all occurrences of it in the project. They can if they want to, but I think this is where we suggest introducing a new readonly variable to expose that data

**Comment by abeizer** [comment]
Resolved every thread except one, which I'm still looking for clarification on: https://github.com/Regression-Games/RegressionDocs/pull/50#discussion_r1496125308

**Review by abeizer — CHANGES_REQUESTED** [review]
forgot to submit earlier comment as blocking

**Review by abeizer — APPROVED** [review]
I'm not sure if we're still waiting on discussion of https://github.com/Regression-Games/RegressionDocs/pull/50#discussion_r1496216555

I'll approve so we can get the rest of these changes in, but I still don't like setting the expectation that our SDK will ask the user to break C# conventions in order to use it. I really think we should change this, even if it's just having them create a public getter 

```
       [RGState]
       public int CurrentHealth
       {
           get { return _currentHp; }
       }
```

This would double as an example where RGState doesn't have a name override.

**Comment by RG-nAmKcAz** [comment]
@vontell  Did you actually try these samples, I don't thing `[RGState]` works on non-public fields or methods ...

**Comment by vontell** [comment]
Went through the docs, fixed one issue, and everything works! Tested in staging.
